src/Grpc.Net.Client/Internal/GrpcCall.cs

```diff
@@ -813,9 +813,12 @@ private void SetFailedResult(Status status)
         // 3. There is an existing activity (to enable activity propagation)
         if (diagnosticSourceEnabled || Logger.IsEnabled(LogLevel.Critical) || Activity.Current != null)
         {
-            activity = new Activity(GrpcDiagnostics.ActivityName);
-            activity.AddTag(GrpcDiagnostics.GrpcMethodTagName, Method.FullName);
-            activity.Start();
+            activity = GrpcDiagnostics.ActivitySource.CreateActivity(GrpcDiagnostics.ActivityName, ActivityKind.Internal);
```

Comment:
Instead of creating the Activity then setting the tags and starting it, you can call StartActivity and passing the tags list using TagList structure.

Reply:
It seems strictly worse than what we're currently doing. I'll leave it as is.